packages/gatsby-theme-carbon/src/components/Code/Code.js

```diff
@@ -17,33 +17,46 @@ import useMetadata from '../../util/hooks/useMetadata';
 const Code = ({ children, className: classNameProp = '', path, src }) => {
   const [hasMoreThanNineLines, setHasMoreThanNineLines] = useState(false);
   const [shouldShowMore, setShouldShowMore] = useState(false);
+  const [isInlineCode, setIsInlineCode] = useState(false);
+  useEffect(() => {
+
+    // Inline code blocks don't have a className prop
+    if (!classNameProp) {
+      setIsInlineCode(true);
+    }
+  }, [classNameProp]);
 
   const { interiorTheme } = useMetadata();
 
   const language = classNameProp.replace(/language-/, '').replace('mdx', 'jsx');
 
   const removeTrailingEmptyLine = (lines) => {
-    const [lastLine] = lines.splice(-1);
-    if (lastLine[0].empty) {
-      return lines;
+    if (lines && lines.length) {
+      const [lastLine] = lines.splice(-1);
+      if (lastLine[0].empty) {
+        return lines;
+      }
+      return [...lines, lastLine];
     }
-    return [...lines, lastLine];
   };
 
   const getLines = (lines) => {
     const withoutTrailingEmpty = removeTrailingEmptyLine(lines);
 
-    if (withoutTrailingEmpty.length > 9) {
+    if (withoutTrailingEmpty && withoutTrailingEmpty.length > 9) {
       setHasMoreThanNineLines(true);
     }
 
     if (shouldShowMore) {
       return withoutTrailingEmpty;
     }
 
-    return withoutTrailingEmpty.slice(0, 9);
+    return withoutTrailingEmpty ? withoutTrailingEmpty.slice(0, 9) : [];
   };
 
+  if (isInlineCode) {
```

Comment:
With this version, inline code are being rendered as pre only and not as <code></code>. This has fixed that. (Maybe temporarily)

Reply:
Add a `// TODO: message...` for this as well, so we can remember to look into and verify this file all together.